jerry-core/jerry.c

```diff
@@ -2090,172 +2094,86 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
-    ecma_value_t completion = vm_run_eval (bytecode_p, false);
-
-    ret_code = jerry_convert_eval_completion_to_retval (completion, retval_p);
-
-    ecma_free_value (completion);
+    ret_val = vm_run_eval (bytecode_p, false);
   }
 
-  return ret_code;
+  return ret_val;
 #else /* !JERRY_ENABLE_SNAPSHOT_EXEC */
-  (void) snapshot_p;
-  (void) snapshot_size;
-  (void) copy_bytecode;
-  (void) retval_p;
+  JERRY_UNUSED (snapshot_p);
+  JERRY_UNUSED (snapshot_size);
+  JERRY_UNUSED (copy_bytecode);
 
-  return JERRY_COMPLETION_CODE_INVALID_SNAPSHOT_VERSION;
+  return ecma_make_simple_value (ECMA_SIMPLE_VALUE_FALSE);
 #endif /* JERRY_ENABLE_SNAPSHOT_EXEC */
 } /* jerry_exec_snapshot */
 
 /**
- * Call ToBoolean operation on the api value.
- *
- * @return true  - if the logical value is true
- *         false - otherwise
+ * @}
  */
-bool
-jerry_value_to_boolean (const jerry_value_t value) /**< input value */
-{
-  jerry_assert_api_available ();
-
-  if (ECMA_IS_VALUE_ERROR (value))
-  {
-    return false;
-  }
-
-  return ecma_op_to_boolean (value);
-} /* jerry_value_to_boolean */
 
 /**
- * Call ToNumber operation on the api value.
- *
- * Note:
- *      - Can throw error, so return value should be checked with 'jerry_value_is_error'.
- *      - Return value must be freed with 'jerry_release_value'.
- *
- * @return number value
+ * ====================== Internal functions ==========================
  */
-jerry_value_t
-jerry_value_to_number (const jerry_value_t value) /**< input value */
-{
-  jerry_assert_api_available ();
-
-  if (ECMA_IS_VALUE_ERROR (value))
-  {
-    return jerry_create_object_value (jerry_create_error (JERRY_ERROR_TYPE, error_value_msg_p));
-  }
-
-  return ecma_op_to_number (value);
-} /* jerry_value_to_number */
 
 /**
- * Call ToObject operation on the api value.
+ * Dispatch call to specified external function using the native handler
  *
  * Note:
- *      - Can throw error, so return value should be checked with 'jerry_value_is_error'.
- *      - Return value must be freed with 'jerry_release_value'.
- *
- * @return object value
- */
-jerry_value_t
-jerry_value_to_object (const jerry_value_t value) /**< input value */
-{
-  jerry_assert_api_available ();
-
-  if (ECMA_IS_VALUE_ERROR (value))
-  {
-    return jerry_create_object_value (jerry_create_error (JERRY_ERROR_TYPE, error_value_msg_p));
-  }
-
-  return ecma_op_to_object (value);
-} /* jerry_value_to_object */
-
-/**
- * Call the ToString ecma builtin operation on the api value.
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
  *
  * Note:
- *      - Can throw error, so return value should be checked with 'jerry_value_is_error'.
- *      - Return value must be freed with 'jerry_release_value'.
+ *      if called native handler returns true, then dispatcher just returns value received
+ *      through 'return value' output argument, otherwise - throws the value as an exception.
```

Comment:
Could this note get cleaned up a bit? Capitalization, `otherwise ` in a new line, etc.